src/Database.m

```diff
@@ -2069,39 +2068,50 @@ -(NSString *)criteriaToSQL:(CriteriaTree *)criteriaTree
 				}
 				
 			case MA_FieldType_Date: {
-				NSCalendarDate * startDate = [NSCalendarDate date];
+				NSDate *startDate = [NSDate date];
 				NSString * criteriaValue = criteria.value.lowercaseString;
-				NSInteger spanOfDays = 1;
+				NSCalendarUnit calendarUnit = NSCalendarUnitDay;
 				
 				// "yesterday" is a short hand way of specifying the previous day.
 				if ([criteriaValue isEqualToString:@"yesterday"])
 				{
-					startDate = [startDate dateByAddingYears:0 months:0 days:-1 hours:0 minutes:0 seconds:0];
+                    startDate = [NSCalendar.currentCalendar dateByAddingUnit:NSCalendarUnitDay
+                                                                       value:-1
+                                                                      toDate:startDate
+                                                                     options:0];
 				}
 				// "last week" is a short hand way of specifying a range from 7 days ago to today.
 				else if ([criteriaValue isEqualToString:@"last week"])
 				{
-					startDate = [startDate dateByAddingYears:0 months:0 days:-6 hours:0 minutes:0 seconds:0];
-					spanOfDays = 7;
+                    startDate = [NSCalendar.currentCalendar dateByAddingUnit:NSCalendarUnitWeekOfMonth
+                                                                       value:-1
+                                                                      toDate:startDate
+                                                                     options:0];
+					calendarUnit = NSCalendarUnitWeekOfMonth;
 				}
-				
-				criteriaValue = [NSString stringWithFormat:@"%ld/%ld/%ld %d:%d:%d", (long)[startDate dayOfMonth], (long)[startDate monthOfYear], (long)[startDate yearOfCommonEra], 0, 0, 0];
-				startDate = [NSCalendarDate dateWithString:criteriaValue calendarFormat:@"%d/%m/%Y %H:%M:%S"];
-				
+
+                startDate = [NSCalendar.currentCalendar startOfDayForDate:startDate];
+
 				if (criteria.operator == MA_CritOper_Is)
 				{
-					NSCalendarDate * endDate;
+					NSDate *endDate;
 
 					// Special case for Date is <date> because the resolution of the date field is in
 					// milliseconds. So we need to translate this to a range for this to make sense.
-					endDate = [startDate dateByAddingYears:0 months:0 days:spanOfDays hours:0 minutes:0 seconds:0];
+                    endDate = [NSCalendar.currentCalendar nextDateAfterDate:startDate
```

Comment:
This appears to be returning `nil` for some reason when I step through the code with a smart folder set with date = yesterday.